airstack-modules/modules/airstack/domain-name/domain-name.ts

```diff
@@ -19,99 +20,71 @@ import {
-   */
+  */
   export function trackDomainOwnerChangedTransaction(
-    blockHeight: BigInt,
-    blockHash: string,
-    blockTimestamp: BigInt,
-    logIndex: BigInt,
-    chainId: string,
-    newOwner: string,
+    block: ethereum.Block,
     transactionHash: string,
-    isMigrated: boolean,
-    node: Bytes,
-    label: Bytes,
+    logOrCallIndex: BigInt,
+    domainId: string,
+    parentDomainId: string,
+    tokenId: string,
+    labelHash: string,
+    labelName: string | null,
+    name: string | null,
+    newOwner: string,
     tokenAddress: string,
-    fromOldRegistry: boolean,
   ): void {
-    let block = getOrCreateAirBlock(chainId, blockHeight, blockHash, blockTimestamp);
-    let domainId = createAirDomainEntityId(node, label);
-    let domain = getOrCreateAirDomain(new Domain(domainId, chainId, block, tokenAddress));
-    let isMigratedMapping = getOrCreateIsMigratedMapping(domainId, block.id, isMigrated);
-    if (fromOldRegistry && isMigratedMapping.isMigrated == true) {
-      // this domain was migrated from the old registry, so we don't need to hanlde old registry event now
-      return;
-    }
+    let chainId = processChainId();
+    let airBlock = getOrCreateAirBlock(chainId, block.number, block.hash.toHexString(), block.timestamp);
+    let domain = getOrCreateAirDomain(new Domain(domainId, chainId, airBlock, tokenAddress));
     let previousOwnerId = domain.owner;
     let parent = getOrCreateAirDomain(new Domain(
-      node.toHexString(),
+      parentDomainId,
       chainId,
-      block,
+      airBlock,
       tokenAddress,
     ));
     if (domain.parent == null && parent != null) {
       parent.subdomainCount = parent.subdomainCount.plus(BIGINT_ONE);
     }
-    if (domain.name == null) {
-      let labelName = ens.nameByHash(label.toHexString());
-      if (labelName != null) {
-        domain.labelName = labelName;
-      }
-      if (labelName === null) {
-        labelName = '[' + label.toHexString().slice(2) + ']';
-      }
-      if (node.toHexString() == ROOT_NODE) {
-        domain.name = labelName;
-      } else {
-        let name = parent.name;
-        if (labelName && name) {
-          domain.name = labelName + '.' + name;
-        }
-      }
-    }
-
-    let tokenId = BigInt.fromUnsignedBytes(label).toString();
-    domain.owner = getOrCreateAirAccount(chainId, newOwner).id;
+    domain.name = name;
+    domain.labelName = labelName;
+    let ownerAccount = getOrCreateAirAccount(chainId, newOwner, airBlock);
+    ownerAccount.save();
+    domain.owner = ownerAccount.id;
     domain.parent = parent.id;
-    domain.labelHash = label;
+    domain.labelHash = labelHash;
     domain.tokenId = tokenId;
-    domain.lastBlock = block.id;
-    recurseSubdomainCountDecrement(domain, chainId, block, tokenAddress);
+    domain.lastBlock = airBlock.id;
+    recurseSubdomainCountDecrement(domain, chainId, airBlock, tokenAddress);
```

Comment:
Found explanation in this issue/pr from the ens subgraph GitHub repo - https://github.com/ensdomains/ens-subgraph/pull/44
this is because when you change the resolver for a domain, your subdomains are no longer linked to the same resolver.
this was also discussed with @0xsarvesh in https://github.com/Airstack-xyz/Subgraphs/pull/34#discussion_r1096798600